design/authenticators/authn_jwt/token_schema.md

```diff
@@ -0,0 +1,180 @@
+# Solution Design - Authn JWT Token Schema
+
+[//]: # "1. Design should be graphical-based and table-based - avoid long text explanations"
+[//]: # "2. Design documents should not be updated after implementation"
+[//]: # "3. Design decisions should be made before writing this document, and as such this document should not include options / choices"
+
+
+## Table of Contents
+[//]: # "You can use this tool to generate a TOC - https://ecotrust-canada.github.io/markdown-toc/"
+
+- [Table of Contents](#table-of-contents)
+- [Glossary](#glossary)
+- [Useful Links](#useful-links)
+- [Requirments](#requirments)
+- [Solution](#solution)
+  * [User Interface](#user-interface)
+    + [Authenticator Policy](#authenticator-policy)
+    + [Variable Values Example](#variable-values-example)
+- [Design](#design)
+  * [Class Diagrams](#class-diagrams)
+  * [Flow Explanation](#flow-explanation)
+- [Backwards Compatibility](#backwards-compatibility)
+- [Affected Components](#affected-components)
+- [Delivery Plan](#delivery-plan)
+- [Delivery Plan](#delivery-plan-1)
+- [Open Questions](#open-questions)
+- [Definition of Done](#definition-of-done)
+- [Solution Review](#solution-review)
+
+<small><i><a href='http://ecotrust-canada.github.io/markdown-toc/'>Table of contents generated with markdown-toc</a></i></small>
+
+## Glossary
+
+[//]: # "Describe terms that will be used throughout the design"
+[//]: # "You can use this tool to generate a table - https://www.tablesgenerator.com/markdown_tables#"
+
+| **Term**         | **Description**                                              |
+| ---------------- | ------------------------------------------------------------ |
+| JWT Claims       | claims are pieces of information asserted about a subject    |
+| Host Annotation  | Defining of check that we should do on authentication request for host. In JWT authentication its a claim check. |
+| Mandatory Claims | JWT Claims that host must validate their existance and value |
+| Claims Mapping   | Mapping between claim name and name of host annotation       |
+| Token Schema     | Combination of mandatory claims and claims mapping           |
+
+## Useful Links
+[//]: # "Add links that may be useful for the reader"
+
+| **Name**   | **Link**                                                     |
+| ---------- | ------------------------------------------------------------ |
+| JWT Claims | https://auth0.com/docs/tokens/json-web-tokens/json-web-token-claims |
+
+## Requirments
+[//]: # "Elaborate on the issue you are writing a solution for"
+
+1. Required fields
+   1. The user will have the ability to declare in the authenticator policy which claims must be in host annotations. And therefore also in the JWT token.
```

Comment:
```suggestion
   1.  In the authenticator policy, the user will have the ability to declare which claims must be in host annotations. And therefore also in the JWT token.
```

Reply:
>  And therefore also in the JWT token.

Not sure what this means. Please explain :)

Reply:
Any annotation in the host that authenticating again JWT authenticator we check it is a claim inside the token payload. If it doesn't we fail the authentication request. I meant here that if we verify that a claim in the host annotations it automatically means that we verify that it is in the decoded token

Reply:
I suggest we make that clear. That the claims are checked inside the token's payload

> if we verify that a claim in the host annotations it automatically means that we verify that it is in the decoded token

great! suggest we add that :)